staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go

```diff
@@ -55,11 +55,15 @@ func newGVKParser(models proto.Models) (*gvkParser, error) {
 	for _, modelName := range models.ListModels() {
 		model := models.LookupModel(modelName)
 		if model == nil {
-			panic("ListModels returns a model that can't be looked-up.")
+			panic(fmt.Sprintf("ListModels returns a model that can't be looked-up for : %v", modelName))
 		}
 		gvkList := parseGroupVersionKind(model)
 		for _, gvk := range gvkList {
 			if len(gvk.Kind) > 0 {
+	                        _, ok := parser.gvks[gvk]
```

Comment:
I still don't know if I want this to panic or return an error. I think panicing would be fine because:
1. This is not user-generated, it's coming from types and openapi generation
2. This is only used for compiled-types, not CRDs.
3. There should never be two models for the same gvk.

In other words, this will only happen if something is very wrong with the apiserver (wrong enough that it's fine if it doesn't boot/function). I'd rather catch error easily when they happen.

Reply:
I don't feel too strongly either way, but fwiw I think either an returning an error here or a panic would both prevent the apiserver from starting up, since this function is called when the handler is registered, right?

Reply:
To me it seems weird to panic when there is an error return value.
I think the panic is a relict from when we silently tolerated errors in calls and the gvkParser was still created on every request. Now that is prevents the API Server from starting (which I guess it should), i'd prefer not to mix errors and panics if not really necessary.

Reply:
What's the consensus on this ?

From above, it seems we have 1.5 versus 1.5 :-)

Reply:
Consensus is that we return an error :-) Sorry I made you change the code @tedyu !